pig/src/test/java/org/jboss/pnc/bacon/pig/impl/addons/scanservice/ValidationTest.java

```diff
@@ -0,0 +1,119 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+
+import java.util.*;
+
+class ValidationTest {
+
+    PostBuildScanService scanService = new PostBuildScanService(null, null, null, null);
+
+    private static final Map<String, ?> benchmarkScanServiceConfigMap = new LinkedHashMap<>();
+    static {
+        benchmarkScanServiceConfigMap.put("productID", null);
+        benchmarkScanServiceConfigMap.put("eventId", null);
+        benchmarkScanServiceConfigMap.put("isManagedService", null);
+        benchmarkScanServiceConfigMap.put("cpaasVersion", null);
+        benchmarkScanServiceConfigMap.put("jobUrl", null);
+        benchmarkScanServiceConfigMap.put("serviceUrl", null);
+        benchmarkScanServiceConfigMap.put("serviceSecretKey", null);
+        benchmarkScanServiceConfigMap.put("serviceSecretValue", null);
+        benchmarkScanServiceConfigMap.put("brewBuilds", null);
+        benchmarkScanServiceConfigMap.put("scmUrl", null);
+        benchmarkScanServiceConfigMap.put("scmRevision", null);
+        benchmarkScanServiceConfigMap.put("extraScmUrls", null);
+        benchmarkScanServiceConfigMap.put("extraBuilds", null);
+        benchmarkScanServiceConfigMap.put("name", null);
+    }
+
+    private static final Map<String, ?> minimalConfigMap = new LinkedHashMap<>();
+    static {
+        minimalConfigMap.put("productID", null);
+        minimalConfigMap.put("serviceUrl", null);
+        minimalConfigMap.put("serviceSecretKey", null);
+        minimalConfigMap.put("serviceSecretValue", null);
+    }
+
+    @Test
+    void positiveKeywordsTest() {
+        showMap(benchmarkScanServiceConfigMap);
+        try {
+            scanService.validateMapKeys(benchmarkScanServiceConfigMap);
+            Assertions.assertTrue(true);
+        } catch (Exception e) {
+            Assertions.fail();
+        }
+    }
+
+    @ParameterizedTest(name = "{index} => validKey={0}, invalidKey={1}")
+    @CsvSource({
+            "productID, product-id",
+            "eventId, eventID",
+            "isManagedService, _isManagedService_",
+            "cpaasVersion, CpaasVersion",
+            "jobUrl, xyz",
+            "serviceUrl, service-url",
+            "serviceSecretKey, serviceSecretkey",
+            "serviceSecretValue, servicesecretValue",
+            "brewBuilds, brewbuilds",
+            "scmUrl, SCMURL",
+            "scmRevision, scm-revision",
+            "extraScmUrls, _xtraScmUrls",
+            "extraBuilds, extraBuilds_" })
+    void negativeKeywordsTest(String validKey, String invalidKey) {
+        HashMap<String, Object> testMap = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        testMap.remove(validKey);
+        testMap.put(invalidKey, null);
+        try {
+            scanService.validateMapKeys(testMap);
+            Assertions.fail();
+        } catch (Exception e) {
+            Assertions.assertTrue(true);
+        }
```

Comment:
```suggestion
	Assertions.assertThrows( Validate.ConfigMissingException.class, () -> scanService.validateMapKeys(testMap) );
        
```

Reply:
Agreed, this is easier to read and catches the specific `ConfigMissingException`, the purpose of the test remains the same.